extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -257,49 +270,84 @@ public static String extractVideoIdFromMixId(final String playlistId) throws Par
 
         } else if (isYoutubeChannelMixId(playlistId)) { // starts with "RMCM"
             // Channel mix are build with RMCM{channelId}, so videoId can't be determined
-            throw new ParsingException("Video id could not be determined from mix id: " + playlistId);
+            throw new ParsingException("Video id could not be determined from mix id: "
+                    + playlistId);
 
         } else if (isYoutubeMixId(playlistId)) { // normal mix, starts with "RD"
             return playlistId.substring(2);
 
         } else { // not a mix
-            throw new ParsingException("Video id could not be determined from mix id: " + playlistId);
+            throw new ParsingException("Video id could not be determined from mix id: "
+                    + playlistId);
         }
     }
 
     public static JsonObject getInitialData(final String html) throws ParsingException {
         try {
             try {
-                final String initialData = Parser.matchGroup1("window\\[\"ytInitialData\"\\]\\s*=\\s*(\\{.*?\\});", html);
+                final String initialData = Parser.matchGroup1(
+                        "window\\[\"ytInitialData\"\\]\\s*=\\s*(\\{.*?\\});", html);
                 return JsonParser.object().from(initialData);
-            } catch (Parser.RegexException e) {
-                final String initialData = Parser.matchGroup1("var\\s*ytInitialData\\s*=\\s*(\\{.*?\\});", html);
+            } catch (final Parser.RegexException e) {
+                final String initialData = Parser.matchGroup1(
+                        "var\\s*ytInitialData\\s*=\\s*(\\{.*?\\});", html);
                 return JsonParser.object().from(initialData);
             }
-        } catch (JsonParserException | Parser.RegexException e) {
+        } catch (final JsonParserException | Parser.RegexException e) {
             throw new ParsingException("Could not get ytInitialData", e);
         }
     }
 
-    public static boolean isHardcodedClientVersionValid() throws IOException, ExtractionException {
-        final String url = "https://www.youtube.com/results?search_query=test&pbj=1";
+    public static boolean areHardcodedClientVersionAndKeyValid() throws IOException,
+            ExtractionException {
+        if (areHardcodedClientVersionAndKeyValidValue != null) {
+            return areHardcodedClientVersionAndKeyValidValue;
+        }
+        // @formatter:off
+        final byte[] body = JsonWriter.string()
+            .object()
+                .object("context")
+                    .object("client")
+                        .value("hl", "en")
+                        .value("gl", "GB")
+                        .value("clientName", "1")
+                        .value("clientVersion", HARDCODED_CLIENT_VERSION)
+                    .end()
+                .end()
+            .end().done().getBytes(UTF_8);
```

Comment:
This could be built manually just by concatenating three strings, can't it? It would improve performance imo, but let's keep this to another PR.